ADALiOS/ADALiOS/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -335,7 +354,7 @@ - (void) validatedAcquireTokenByRefreshToken:(NSString*)refreshToken
           additionalHeaders:nil
                  completion:^(NSDictionary *response)
      {
-         ADTokenCacheStoreItem* resultItem = (cacheItem) ? cacheItem : [ADTokenCacheStoreItem new];
+         ADTokenCacheStoreItem* resultItem = (cacheItem) ? [ADTokenCacheStoreItem initWithStoreItem:cacheItem] : [ADTokenCacheStoreItem new];
```

Comment:
Agree :)
Now the retry behaviour has been implemented at the web request level rather than the authentication request level.
But this line for fixing the infinite-retry bug is still needed, otherwise infinite retry still happens :P